doc/code/qml_pulse.rst

```diff
@@ -0,0 +1,215 @@
+qml.pulse
+=========
+
+.. automodule:: pennylane.pulse
```

Comment:
Part of rearranging this was to make it less of a walk-through and more of a reference guide. For this reason, we want the list of classes/functions first; this is also how the other Module API pages are. 

We could add a brief (1-2 paragraph) introduction before the function tables, but the function overview should be at the top, and usage details follow.

Reply:
I like the idea of having some text before the functions, otherwise one might mistake the module page for a pure listing of functions and not scroll down

Reply:
Cool, I'll add something like [this](https://docs.pennylane.ai/en/stable/code/qml_gradients.html)